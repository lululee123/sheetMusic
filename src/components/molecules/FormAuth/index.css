.form-auth {
	display: flex;
	position: relative;
	flex-direction: column;
	width: 400px;
	margin: 80px auto 0 auto;

	@media screen and (--mobile) {
		width: 100%;
		margin-top: 0;
	}

	& .logo {
		@media screen and (--mobile) {
			margin: -4px auto 0 auto;
		}
	}
}

.form-basic {
	display: block;
}

.form-signup {
	& .title {
		font-size: 32px;
		font-weight: bold;
		line-height: 1.25;
		color: var('--darkBlack');
		margin-top: 49px;

		@media screen and (--mobile) {
			margin-top: 52px;
		}
	}

	& > .buttons {
		display: flex;
		justify-content: space-between;
		margin-top: 24px;

		& .social-button {
			width: calc(50% - 4px);
			height: 48px;
		}

		& .text {
			font-size: 14px;
			font-weight: 500;
			line-height: 1.71;
			height: 40px;
			padding: 0;
			min-width: 32px;
		}
	}

	& .singel-btn {
		width: 100%;

		& .text {
			font-size: 14px;
			font-weight: 500;
			line-height: 1.71;
			height: 40px;
			padding: 0;
		}
	}

	& > .division {
		display: flex;
		align-items: center;
		margin-top: 24px;

		& > span {
			margin: 0 16px;
			font-size: 14px;
			font-weight: 500;
			line-height: 1.71;
			text-align: center;
			color: var(--darkGray);
		}
	}

	& .big-margin {
		margin-top: 24px;
	}

	& .small-margin {
		margin-top: 16px;
	}

	& .password-box {
		display: flex;
		justify-content: space-between;
		margin-top: 16px;

		& > div {
			width: calc(50% - 8px);
		}
	}
}

.form-signup-footer {
	margin-top: 48px;
	font-size: 14px;
	font-weight: 500;
	line-height: 1.57;
	color: var('--darkBlack');

	@media screen and (--mobile) {
		margin-top: 111px;
		text-align: center;
	}

	& .button {
		color: var('--linkBlue');
		text-decoration: none;
		outline: none;
		background-color: transparent;
		border: none;
		padding: 0;
		font-size: 14px;
		font-weight: 500;
		line-height: 1.57;
		cursor: pointer;
	}
}

.form-login-footer {
	margin-top: 48px;

	@media screen and (--mobile) {
		margin-top: 49px;
		text-align: center;
	}

	& .button {
		color: var('--linkBlue');
		text-decoration: none;
		outline: none;
		background-color: transparent;
		border: none;
		padding: 0;
		font-size: 14px;
		font-weight: 500;
		line-height: 1.57;
		cursor: pointer;
	}

	& .login-foot-margin {
		margin-top: 12px;
	}

	& .item {
		font-size: 14px;
		font-weight: 500;
		line-height: 1.57;
		color: var('--darkBlack');
	}
}
