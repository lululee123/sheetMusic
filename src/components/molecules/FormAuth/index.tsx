import React, { useState, useEffect, ReactElement } from 'react';
import classnames from 'classnames';
import { useTranslation } from 'react-i18next';

import { emailValid, passwordValid } from 'util/helper';

import ButtonGoogle from 'components/atoms/ButtonGoogle';
import ButtonFacebook from 'components/atoms/ButtonFacebook';
import Division from 'components/atoms/Division';
import InputField from 'components/atoms/InputField';
import Button from 'components/atoms/Button';
import Link from 'components/atoms/Link';

import styles from './index.css';

interface FormBasicProperty {
	contentClassName?: string;
	footerClassName?: string;
	renderContent: () => null | ReactElement;
	renderFooter?: () => null | ReactElement;
}

const FormBasic: React.FC<FormBasicProperty> = ({
	contentClassName,
	footerClassName,
	renderContent = () => null,
	renderFooter = () => null,
}) => (
	<div className={styles.formBasic}>
		<div className={classnames(styles.content, contentClassName)}>{renderContent()}</div>
		{typeof renderFooter !== 'undefined' && (
			<div className={classnames(styles.footer, footerClassName)}>{renderFooter()}</div>
		)}
	</div>
);

interface FormProperty {

}

interface FormLoginProperty extends FormProperty {

}

const FormLogin: React.FC<FormLoginProperty> = ({
	
}) => {
	const { t } = useTranslation(['auth']);
	const [emailError, setEmailError] = useState(false);
	const checkLogin = () => {
		if (!emailValid(auth.loginForm.email)) {
			setEmailError(true);
		} else {
			Login();
		}
	};

	return (
		<FormBasic
			contentClassName={styles.formSignup}
			renderContent={() => (
				<>
					<div className={styles.title}>{t('login')}</div>
					<div className={styles.buttons}>
						<ButtonGoogle
							className={styles.socialButton}
							text={`Google ${t('login')}`}
							onClick={loginGoogle}
						/>
						<ButtonFacebook
							className={styles.socialButton}
							text={`Facebook ${t('login')}`}
							onClick={loginFacebook}
						/>
					</div>
					<div className={styles.division}>
						<Division />
						<span>{t('or')}</span>
						<Division />
					</div>
					<InputField
						className={styles.bigMargin}
						title={t('account')}
						placeholder={t('loginAccountPlaceholder')}
						type="text"
						value={auth.loginForm.email}
						onChange={e => {
							updateForm({ type: 'loginForm', key: 'email', value: e.target.value });
							setEmailError(false);
						}}
						mustFill
						error={emailError}
						errorText={t('mailFormatError')}
					/>
					<InputField
						className={styles.smallMargin}
						title={t('password')}
						placeholder={t('loginPasswordPlaceholder')}
						type="password"
						value={auth.loginForm.password}
						onChange={e => {
							updateForm({ type: 'loginForm', key: 'password', value: e.target.value });
						}}
						mustFill
					/>
					<Button
						className={classnames(styles.singelBtn, styles.bigMargin)}
						textClassName={styles.text}
						feather
						onClick={checkLogin}
						disabled={!(auth.loginForm.email && auth.loginForm.password)}
					>
						{t('login')}
					</Button>
				</>
			)}
			footerClassName={styles.formLoginFooter}
			renderFooter={() => (
				<>
					<div className={styles.item}>
						<button
							onClick={() => {
								openModal({ category: 'auth', type: 'forget' });
								closeModal({ category: 'alert' });
							}}
							type="button"
						>
							{`${t('forget')}?`}
						</button>
					</div>
					<div className={classnames(styles.loginFootMargin, styles.item)}>
						{`${t('noAccount')} `}
						<button
							onClick={() => {
								openModal({ category: 'auth', type: 'signup' });
								closeModal({ category: 'alert' });
							}}
							type="button"
						>
							{t('signup')}
						</button>
					</div>
				</>
			)}
		/>
	);
};

interface FormSignupProperty extends FormProperty {
	Signup: () => void;
	openModal: (p: OpenModalActionState) => void;
	closeModal: (p: CloseModalActionState) => void;
	loginGoogle: () => void;
	loginFacebook: () => void;
}

const FormSignup: React.FC<FormSignupProperty> = ({
	updateForm,
	auth,
	Signup,
	openModal,
	closeModal,
	loginGoogle,
	loginFacebook,
}) => {
	const { t } = useTranslation(['auth']);
	const [emailError, setEmailError] = useState(false);
	const [passwordError, setPasswordError] = useState({ status: false, text: '' });

	const checkSignup = () => {
		if (!emailValid(auth.signupForm.email)) {
			setEmailError(true);
		} else if (!passwordValid(auth.signupForm.password)) {
			setPasswordError({ status: true, text: `${t('passwordFormatError')}` });
		} else if (auth.signupForm.password !== auth.signupForm.repeatPassword) {
			setPasswordError({ status: true, text: `${t('notSame')}` });
		} else {
			Signup();
		}
	};

	return (
		<FormBasic
			contentClassName={styles.formSignup}
			renderContent={() => (
				<>
					<div className={styles.title}>{t('signup')}</div>
					<div className={styles.buttons}>
						<ButtonGoogle
							className={styles.socialButton}
							text={`Google ${t('signup')}`}
							onClick={loginGoogle}
						/>
						<ButtonFacebook
							className={styles.socialButton}
							text={`Facebook ${t('signup')}`}
							onClick={loginFacebook}
						/>
					</div>
					<div className={styles.division}>
						<Division />
						<span>{t('or')}</span>
						<Division />
					</div>
					<InputField
						className={styles.bigMargin}
						title={t('account')}
						placeholder={t('accountPlaceholder')}
						type="text"
						value={auth.signupForm.email}
						onChange={e => {
							updateForm({ type: 'signupForm', key: 'email', value: e.target.value });
							setEmailError(false);
						}}
						mustFill
						error={emailError}
						errorText={t('mailFormatError')}
					/>
					<div className={styles.passwordBox}>
						<InputField
							title={t('password')}
							placeholder={t('passwordPlaceholder')}
							type="password"
							value={auth.signupForm.password}
							onChange={e => {
								updateForm({ type: 'signupForm', key: 'password', value: e.target.value });
								setPasswordError({ status: false, text: '' });
							}}
							error={passwordError.status}
							errorText={passwordError.status ? passwordError.text : ''}
							mustFill
						/>
						<InputField
							title={t('confirmPassword')}
							placeholder={t('confirmPasswordPlaceholder')}
							type="password"
							value={auth.signupForm.repeatPassword}
							onChange={e => {
								updateForm({ type: 'signupForm', key: 'repeatPassword', value: e.target.value });
								setPasswordError({ status: false, text: '' });
							}}
							mustFill
						/>
					</div>
					<Button
						className={classnames(styles.singelBtn, styles.bigMargin)}
						textClassName={styles.text}
						feather
						onClick={checkSignup}
						disabled={
							!(auth.signupForm.email && auth.signupForm.password && auth.signupForm.repeatPassword)
						}
					>
						{t('signup')}
					</Button>
				</>
			)}
			footerClassName={styles.formSignupFooter}
			renderFooter={() => (
				<div className={styles.item}>
					{`${t('hasAccount')} `}
					<button
						onClick={() => {
							openModal({ category: 'auth', type: 'login' });
							closeModal({ category: 'alert' });
						}}
						type="button"
					>
						{t('login')}
					</button>
				</div>
			)}
		/>
	);
};

interface FormResetPasswordProperty extends FormProperty {
	Reset: () => void;
}

const FormResetPassword: React.FC<FormResetPasswordProperty> = ({ updateForm, auth, Reset }) => {
	const { t } = useTranslation(['auth']);
	const [passwordError, setPasswordError] = useState({ status: false, text: '' });
	const checkReset = () => {
		if (!passwordValid(auth.resetForm.newPassword)) {
			setPasswordError({ status: true, text: `${t('passwordFormatError')}` });
		} else if (auth.resetForm.newPassword !== auth.resetForm.repeatPassword) {
			setPasswordError({ status: true, text: `${t('notSame')}` });
		} else {
			Reset();
		}
	};

	return (
		<FormBasic
			contentClassName={styles.formSignup}
			renderContent={() => (
				<>
					<div className={styles.title}>{t('reset')}</div>
					<InputField
						className={styles.bigMargin}
						title={t('resetPassword')}
						placeholder={t('passwordPlaceholder')}
						type="password"
						value={auth.resetForm.newPassword}
						onChange={e => {
							updateForm({ type: 'resetForm', key: 'newPassword', value: e.target.value });
							setPasswordError({ status: false, text: '' });
						}}
						error={passwordError.status}
						errorText={passwordError.text}
						mustFill
					/>
					<InputField
						className={styles.smallMargin}
						title={t('confirmPassword')}
						placeholder={t('confirmPasswordPlaceholder')}
						type="password"
						value={auth.resetForm.repeatPassword}
						onChange={e => {
							updateForm({ type: 'resetForm', key: 'repeatPassword', value: e.target.value });
							setPasswordError({ status: false, text: '' });
						}}
						mustFill
					/>
					<Button
						className={classnames(styles.singelBtn, styles.bigMargin)}
						textClassName={styles.text}
						feather
						onClick={checkReset}
						disabled={!(auth.resetForm.newPassword && auth.resetForm.repeatPassword)}
					>
						{t('confirm')}
					</Button>
				</>
			)}
		/>
	);
};

interface FormForgetProperty extends FormProperty {
	Forget: () => any; // eslint-disable-line @typescript-eslint/no-explicit-any
}

const FormForget: React.FC<FormForgetProperty> = ({ auth, updateForm, Forget }) => {
	const { t } = useTranslation(['auth']);
	const [emailError, setEmailError] = useState(false);
	const [btnDisable, setBtnDisable] = useState(false);

	const checkForget = async () => {
		if (!emailValid(auth.forgotForm.email)) {
			setEmailError(true);
		} else {
			const {
				value: { success },
			} = await Forget();

			// if forget email send successfully, set button disable
			if (success) {
				setBtnDisable(true);
			}
		}
	};

	useEffect(() => {
		let timeout: number;

		if (btnDisable) {
			timeout = window.setTimeout(() => {
				setBtnDisable(false);
			}, 180000);
		}

		return () => window.clearTimeout(timeout);
	}, [btnDisable]);

	return (
		<FormBasic
			contentClassName={styles.formSignup}
			renderContent={() => (
				<>
					<div className={styles.title}>{t('forget')}</div>
					<InputField
						className={styles.bigMargin}
						title={t('mail')}
						placeholder={t('forgetMailPlaceholder')}
						type="text"
						value={auth.forgotForm.email}
						onChange={e => {
							updateForm({ type: 'forgotForm', key: 'email', value: e.target.value });
							setEmailError(false);
						}}
						error={emailError}
						errorText={t('mailFormatError')}
						mustFill
					/>
					<Button
						className={classnames(styles.singelBtn, styles.bigMargin)}
						textClassName={styles.text}
						feather
						onClick={checkForget}
						disabled={!auth.forgotForm.email ? true : !!btnDisable}
					>
						{t('send')}
					</Button>
				</>
			)}
		/>
	);
};

interface FormAuthProperty {
	className?: string;
	type: string;
}

const FormAuth: React.FC<FormAuthProperty> = ({
	className,
	type: PropsType,
}) => {
	return (
		<div className={classnames(styles.formAuth, className)}>
			{PropsType === 'login' && (
				<FormLogin />
			)}
			{/* {PropsType === 'signup' && (
				<FormSignup
					updateForm={updateForm}
					auth={auth}
					Signup={Signup}
					openModal={openModal}
					closeModal={closeModal}
					loginFacebook={loginFacebook}
					loginGoogle={loginGoogle}
				/>
			)}
			{PropsType === 'forget' && <FormForget updateForm={updateForm} auth={auth} Forget={Forget} />} */}
		</div>
	);
};

export default FormAuth;
