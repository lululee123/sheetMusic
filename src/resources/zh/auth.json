{
	"signup": "註冊",
	"or": "或",
	"mustFill": "必填",
	"account": "帳號",
	"accountPlaceholder": "請填入 email",
	"password": "密碼",
	"passwordPlaceholder": "請填入 8 - 12 碼英數",
	"confirmPassword": "再次確認密碼",
	"confirmPasswordPlaceholder": "再次輸入密碼",
	"back": "返回",
	"hasAccount": "已經有帳號了嗎？從這裡",
	"login": "登入",
	"loginAccountPlaceholder": "請填入註冊的 email",
	"loginPasswordPlaceholder": "請填入密碼",
	"forget": "忘記密碼",
	"noAccount": "還沒有帳號嗎？從這裡",
	"mail": "信箱",
	"forgetMailPlaceholder": "請輸入註冊信箱，系統將會寄送密碼重設連結給您",
	"send": "發送",
	"reset": "重設密碼",
	"resetPassword": "輸入新密碼",
	"resetConfirmPassword": "再次確認新密碼",
	"confirm": "確認",
	"mailFormatError": "請輸入正確的信箱格式",
	"passwordFormatError": "請輸入正確的密碼格式",
	"notSame": "密碼不相符"
}
