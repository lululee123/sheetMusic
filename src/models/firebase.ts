import { createAction, handleActions, Action } from 'redux-actions';
import { useRedux } from 'util/hook/redux';

import firebase from 'firebase';
import config from '../util/firebase';

import { State as GlobalState } from './reducers';

// For Global State usage
export interface State {
	loading: boolean;
	data: { [props: string]: string | number | boolean };
}

const initialState: State = {
	loading: true,
	data: {},
};

export interface AuthTokenActionState {
	token: string;
}

export const loadAuthToken = createAction('LOAD_AUTH_TOKEN', (token: string) => ({
	token,
}));

export const initFirebase = createAction('INIT_FIREBASE', async () => {
	const database = firebase.initializeApp(config).database();
	const data = await database.ref('/test').once('value', e => {
		console.log(e.val());
		return e.val();
	});
	return { data };
});

export const reducer = {
	firebase: handleActions<State, any>( // eslint-disable-line @typescript-eslint/no-explicit-any
		{
			INIT_FIREBASE_FULFILLED: (state, action) => ({
				...state,
				loading: false,
				data: action.payload.data,
			}),

			LOAD_AUTH_TOKEN: (state, action: Action<AuthTokenActionState>) => ({
				...state,

				token: action.payload.token,
			}),
		},
		initialState,
	),
};

const mapHooksToState = (state: GlobalState) => ({
	firebase: state.firebase,
});

const firebaseActionsMap = {
	initFirebase,
};

type FirebaseSelector = ReturnType<typeof mapHooksToState>;
type FirebaseActionsMap = typeof firebaseActionsMap;

export const useFirebase = () =>
	useRedux<FirebaseSelector, FirebaseActionsMap>(mapHooksToState, firebaseActionsMap);
