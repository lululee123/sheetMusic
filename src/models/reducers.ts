import { combineReducers } from 'redux';

import * as routing from './routing';
import * as firebase from './firebase';
import * as i18n from './i18n';

// For Global State interface
export interface State {
	routing: routing.State;
	firebase: firebase.State;
	i18n: i18n.State;
}

const reducers = combineReducers<State>({
	...routing.reducer,
	...firebase.reducer,
	...i18n.reducer,
});

export default reducers;
