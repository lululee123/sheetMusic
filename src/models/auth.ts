import { createAction, handleActions, Action } from 'redux-actions';
import { useRedux } from 'util/hook/redux';
import { Dispatch } from 'redux';
import firebase from 'firebase';

import storage from 'util/storage';
import { isExist } from 'util/helper';

import { openModal } from './modal';

import { GetState, State as GlobalState } from './reducers';
import history from '../store/history';

export interface AuthTokenActionState {
	token: string;
}

export interface GetDataActionState {
	data: { [props: string]: string | number | boolean };
}

export const getData = createAction(
	'GET_DATA',
	(data: { [props: string]: string | number | boolean }) => ({ data }),
);

export const saveTokenAndGetUserData = createAction(
	'SAVE_ACCESS_TOKEN',
	(token: string) => async (dispatch: Dispatch) => {
		storage.setItem('token', JSON.stringify(token));
		const data = firebase.database().ref(`users/${token}`);
		data.on('value', (e: { val: () => { [props: string]: string | number | boolean } }) => {
			dispatch(e.val() === null ? getData({}) : getData(e.val()));
		});

		return { token };
	},
);

export interface UpdateFormActionState {
	type: string;
	key: string;
	value: string;
}

export const updateForm = createAction<UpdateFormActionState, UpdateFormActionState>(
	'UPDATE_FORM',
	({ type, key, value }) => ({
		type,
		key,
		value,
	}),
);

export const login = createAction('LOGIN', () => async (dispatch: Dispatch, getState: GetState) => {
	const {
		auth: {
			loginForm: { email, password },
		},
	} = getState();

	return firebase
		.auth()
		.signInWithEmailAndPassword(email, password)
		.then(async data => {
			await dispatch(saveTokenAndGetUserData(data.user?.uid || ''));
			history.push('/');
		})
		.catch(error => {
			dispatch(openModal({ category: 'alert', type: error.code }));
		});
});

export const signup = createAction(
	'SIGNUP',
	() => async (dispatch: Dispatch, getState: GetState) => {
		const {
			auth: {
				signupForm: { email, password },
			},
		} = getState();

		return firebase
			.auth()
			.createUserWithEmailAndPassword(email, password)
			.then(async data => {
				await dispatch(saveTokenAndGetUserData(data.user?.uid || ''));
				history.push('/');
			})
			.catch(error => {
				dispatch(openModal({ category: 'alert', type: error.code }));
			});
	},
);

export const forget = createAction(
	'FORGET',
	() => async (dispatch: Dispatch, getState: GetState) => {
		const {
			auth: {
				forgetForm: { email },
			},
		} = getState();

		return firebase
			.auth()
			.sendPasswordResetEmail(email, null)
			.then(() => {
				dispatch(openModal({ category: 'alert', type: 'resetMailSend' }));
			})
			.catch(() => {
				dispatch(openModal({ category: 'alert', type: 'resetMailSendFail' }));
			});
	},
);

export const loginWithGoogle = createAction(
	'LOGIN_WITH_GOOGLE',
	() => async (dispatch: Dispatch) => {
		const provider = new firebase.auth.GoogleAuthProvider();
		firebase.auth().useDeviceLanguage();

		return firebase
			.auth()
			.signInWithPopup(provider)
			.then(async () => {
				await dispatch(saveTokenAndGetUserData(firebase.auth().currentUser?.uid || ''));
				history.push('/');
			})
			.catch(() => {
				dispatch(openModal({ category: 'alert', type: 'fail' }));
			});
	},
);

export const loginWithFaceBook = createAction(
	'LOGIN_WITH_FACEBOOK',
	() => async (dispatch: Dispatch) => {
		const provider = new firebase.auth.FacebookAuthProvider();
		firebase.auth().useDeviceLanguage();

		return firebase
			.auth()
			.signInWithPopup(provider)
			.then(async () => {
				await dispatch(saveTokenAndGetUserData(firebase.auth().currentUser?.uid || ''));
				history.push('/');
			})
			.catch(() => {
				dispatch(openModal({ category: 'alert', type: 'fail' }));
			});
	},
);

export const clearTokenData = createAction('CLEAR_TOKEN_DATA');

const logout = createAction('LOGOUT', () => async (dispatch: Dispatch) => {
	storage.removeItem('token');
	dispatch(clearTokenData());
	firebase.auth().signOut();
});

// For Global State usage
export interface State {
	[key: string]: {};
	loginForm: {
		email: string;
		password: string;
	};

	signupForm: {
		email: string;
		password: string;
		repeatPassword: string;
	};

	forgetForm: {
		email: string;
	};
	token: string;
	userInfo: {};
	data: { list: [{ [props: string]: string | number | boolean }] };
}

export const initialState: State = {
	loginForm: {
		email: '',
		password: '',
	},

	signupForm: {
		email: '',
		password: '',
		repeatPassword: '',
	},

	forgetForm: {
		email: '',
	},
	token: '',
	userInfo: {},
	data: {},
};

export const reducer = {
	auth: handleActions<State, any>( // eslint-disable-line @typescript-eslint/no-explicit-any
		{
			SAVE_ACCESS_TOKEN: (state, action: Action<AuthTokenActionState>) => ({
				...state,

				token: action.payload.token,
			}),

			UPDATE_FORM: (state, action: Action<UpdateFormActionState>) => ({
				...state,
				[action.payload.type]: {
					...state[action.payload.type],
					[action.payload.key]: action.payload.value,
				},
			}),

			GET_DATA: (state, action: Action<GetDataActionState>) => ({
				...state,

				data: action.payload.data,
			}),

			CLEAR_TOKEN_DATA: state => ({
				...state,

				token: '',
			}),
		},
		initialState,
	),
};

const mapHooksToState = (state: GlobalState) => ({
	auth: state.auth,
	isLogin: isExist(state.auth.token),
});

const authActionsMap = {
	updateForm,
	logout,
	login,
	signup,
	forget,
	loginWithGoogle,
	loginWithFaceBook,
};

type AuthSelector = ReturnType<typeof mapHooksToState>;
type AuthActionsMap = typeof authActionsMap;

export const useAuth = () =>
	useRedux<AuthSelector, AuthActionsMap>(mapHooksToState, authActionsMap);
