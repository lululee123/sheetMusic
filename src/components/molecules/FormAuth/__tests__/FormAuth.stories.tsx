import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import FormAuth from 'components/molecules/FormAuth';

import { initialState } from 'models/auth';

const stories = storiesOf('molecules/FormAuth', module);

stories.add('with type login', () => (
	<FormAuth
		type="login"
		openModal={action('openModal')}
		closeModal={action('closeModal')}
		updateForm={action('updateForm')}
		Signup={action('Signup')}
		loginFacebook={action('loginFacebook')}
		loginGoogle={action('loginGoogle')}
		Login={action('Login')}
		Forget={action('Forget')}
		auth={initialState}
	/>
));

stories.add('with type signup', () => (
	<FormAuth
		type="signup"
		openModal={action('openModal')}
		closeModal={action('closeModal')}
		updateForm={action('updateForm')}
		Signup={action('Signup')}
		loginFacebook={action('loginFacebook')}
		loginGoogle={action('loginGoogle')}
		Login={action('Login')}
		Forget={action('Forget')}
		auth={initialState}
	/>
));

stories.add('with type reset password', () => (
	<FormAuth
		type="resetPassword"
		openModal={action('openModal')}
		closeModal={action('closeModal')}
		updateForm={action('updateForm')}
		Signup={action('Signup')}
		loginFacebook={action('loginFacebook')}
		loginGoogle={action('loginGoogle')}
		Login={action('Login')}
		Forget={action('Forget')}
		auth={initialState}
	/>
));

stories.add('with type verify', () => (
	<FormAuth
		type="verify"
		openModal={action('openModal')}
		closeModal={action('closeModal')}
		updateForm={action('updateForm')}
		Signup={action('Signup')}
		loginFacebook={action('loginFacebook')}
		loginGoogle={action('loginGoogle')}
		Login={action('Login')}
		Forget={action('Forget')}
		auth={initialState}
	/>
));

stories.add('with type forget', () => (
	<FormAuth
		type="forget"
		openModal={action('openModal')}
		closeModal={action('closeModal')}
		updateForm={action('updateForm')}
		Signup={action('Signup')}
		loginFacebook={action('loginFacebook')}
		loginGoogle={action('loginGoogle')}
		Login={action('Login')}
		Forget={action('Forget')}
		auth={initialState}
	/>
));
