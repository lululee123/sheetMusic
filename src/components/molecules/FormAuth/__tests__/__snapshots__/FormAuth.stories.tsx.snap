// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with type forget 1`] = `
<DocumentFragment>
  <div
    class="formAuth"
  >
    <a
      class="logo"
      href="/"
      role="button"
      tabindex="0"
    >
      <test-file-stub />
    </a>
    <div
      class="formBasic"
    >
      <div
        class="content formSignup"
      >
        <div
          class="title"
        >
          forget
        </div>
        <div
          class="inputField bigMargin"
        >
          <div
            class="top"
          >
            <span
              class="title"
            >
              mail
            </span>
            <span
              class="mustFill"
            >
              mustFill
            </span>
          </div>
          <input
            class="input"
            placeholder="forgetMailPlaceholder"
            type="text"
            value=""
          />
        </div>
        <button
          class="button singelBtn bigMargin disabled feather normal blue"
          type="button"
        >
          <div
            class="featherIcon disabled normal"
            style="transform: rotate(180deg);"
          >
            <test-file-stub />
          </div>
          <div
            class="main text"
          >
            <span>
              send
            </span>
          </div>
          <div
            class="featherIcon disabled normal"
            style="transform: rotate(0deg);"
          >
            <test-file-stub />
          </div>
        </button>
      </div>
      <div
        class="footer"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`with type login 1`] = `
<DocumentFragment>
  <div
    class="formAuth"
  >
    <a
      class="logo"
      href="/"
      role="button"
      tabindex="0"
    >
      <test-file-stub />
    </a>
    <div
      class="formBasic"
    >
      <div
        class="content formSignup"
      >
        <div
          class="title"
        >
          login
        </div>
        <div
          class="buttons"
        >
          <button
            class="button buttonGoogle socialButton"
            type="button"
          >
            <test-file-stub />
            <div
              class="main"
            >
              <span>
                Google login
              </span>
            </div>
          </button>
          <button
            class="button buttonFacebook socialButton"
            type="button"
          >
            <test-file-stub />
            <div
              class="main"
            >
              <span>
                Facebook login
              </span>
            </div>
          </button>
        </div>
        <div
          class="division"
        >
          <div
            class="division"
          />
          <span>
            or
          </span>
          <div
            class="division"
          />
        </div>
        <div
          class="inputField bigMargin"
        >
          <div
            class="top"
          >
            <span
              class="title"
            >
              account
            </span>
            <span
              class="mustFill"
            >
              mustFill
            </span>
          </div>
          <input
            class="input"
            placeholder="loginAccountPlaceholder"
            type="text"
            value=""
          />
        </div>
        <div
          class="inputField smallMargin"
        >
          <div
            class="top"
          >
            <span
              class="title"
            >
              password
            </span>
            <span
              class="mustFill"
            >
              mustFill
            </span>
          </div>
          <input
            class="input"
            placeholder="loginPasswordPlaceholder"
            type="password"
            value=""
          />
        </div>
        <button
          class="button singelBtn bigMargin disabled feather normal blue"
          type="button"
        >
          <div
            class="featherIcon disabled normal"
            style="transform: rotate(180deg);"
          >
            <test-file-stub />
          </div>
          <div
            class="main text"
          >
            <span>
              login
            </span>
          </div>
          <div
            class="featherIcon disabled normal"
            style="transform: rotate(0deg);"
          >
            <test-file-stub />
          </div>
        </button>
      </div>
      <div
        class="footer formLoginFooter"
      >
        <div
          class="item"
        >
          <button
            type="button"
          >
            forget?
          </button>
        </div>
        <div
          class="loginFootMargin item"
        >
          noAccount 
          <button
            type="button"
          >
            signup
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`with type reset password 1`] = `
<DocumentFragment>
  <div
    class="formAuth"
  >
    <a
      class="logo"
      href="/"
      role="button"
      tabindex="0"
    >
      <test-file-stub />
    </a>
    <div
      class="formBasic"
    >
      <div
        class="content formSignup"
      >
        <div
          class="title"
        >
          reset
        </div>
        <div
          class="inputField bigMargin"
        >
          <div
            class="top"
          >
            <span
              class="title"
            >
              resetPassword
            </span>
            <span
              class="mustFill"
            >
              mustFill
            </span>
          </div>
          <input
            class="input"
            placeholder="passwordPlaceholder"
            type="password"
            value=""
          />
        </div>
        <div
          class="inputField smallMargin"
        >
          <div
            class="top"
          >
            <span
              class="title"
            >
              confirmPassword
            </span>
            <span
              class="mustFill"
            >
              mustFill
            </span>
          </div>
          <input
            class="input"
            placeholder="confirmPasswordPlaceholder"
            type="password"
            value=""
          />
        </div>
        <button
          class="button singelBtn bigMargin disabled feather normal blue"
          type="button"
        >
          <div
            class="featherIcon disabled normal"
            style="transform: rotate(180deg);"
          >
            <test-file-stub />
          </div>
          <div
            class="main text"
          >
            <span>
              confirm
            </span>
          </div>
          <div
            class="featherIcon disabled normal"
            style="transform: rotate(0deg);"
          >
            <test-file-stub />
          </div>
        </button>
      </div>
      <div
        class="footer"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`with type signup 1`] = `
<DocumentFragment>
  <div
    class="formAuth"
  >
    <a
      class="logo"
      href="/"
      role="button"
      tabindex="0"
    >
      <test-file-stub />
    </a>
    <div
      class="formBasic"
    >
      <div
        class="content formSignup"
      >
        <div
          class="title"
        >
          signup
        </div>
        <div
          class="buttons"
        >
          <button
            class="button buttonGoogle socialButton"
            type="button"
          >
            <test-file-stub />
            <div
              class="main"
            >
              <span>
                Google signup
              </span>
            </div>
          </button>
          <button
            class="button buttonFacebook socialButton"
            type="button"
          >
            <test-file-stub />
            <div
              class="main"
            >
              <span>
                Facebook signup
              </span>
            </div>
          </button>
        </div>
        <div
          class="division"
        >
          <div
            class="division"
          />
          <span>
            or
          </span>
          <div
            class="division"
          />
        </div>
        <div
          class="inputField bigMargin"
        >
          <div
            class="top"
          >
            <span
              class="title"
            >
              account
            </span>
            <span
              class="mustFill"
            >
              mustFill
            </span>
          </div>
          <input
            class="input"
            placeholder="accountPlaceholder"
            type="text"
            value=""
          />
        </div>
        <div
          class="passwordBox"
        >
          <div
            class="inputField"
          >
            <div
              class="top"
            >
              <span
                class="title"
              >
                password
              </span>
              <span
                class="mustFill"
              >
                mustFill
              </span>
            </div>
            <input
              class="input"
              placeholder="passwordPlaceholder"
              type="password"
              value=""
            />
          </div>
          <div
            class="inputField"
          >
            <div
              class="top"
            >
              <span
                class="title"
              >
                confirmPassword
              </span>
              <span
                class="mustFill"
              >
                mustFill
              </span>
            </div>
            <input
              class="input"
              placeholder="confirmPasswordPlaceholder"
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          class="button singelBtn bigMargin disabled feather normal blue"
          type="button"
        >
          <div
            class="featherIcon disabled normal"
            style="transform: rotate(180deg);"
          >
            <test-file-stub />
          </div>
          <div
            class="main text"
          >
            <span>
              signup
            </span>
          </div>
          <div
            class="featherIcon disabled normal"
            style="transform: rotate(0deg);"
          >
            <test-file-stub />
          </div>
        </button>
      </div>
      <div
        class="footer formSignupFooter"
      >
        <div
          class="item"
        >
          hasAccount 
          <button
            type="button"
          >
            login
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`with type verify 1`] = `
<DocumentFragment>
  <div
    class="formAuth"
  >
    <a
      class="logo"
      href="/"
      role="button"
      tabindex="0"
    >
      <test-file-stub />
    </a>
  </div>
</DocumentFragment>
`;
